browser-extension/mv2/src/background/background.js

```diff
@@ -809,22 +809,28 @@ BG.Methods.sendMessageToAllAppTabs = function (messageObject, callback) {
       console.log("DefaultHandler: Sending Message to Runtime: ", messageObject);
     };
 
-  chrome.tabs.query({ url: RQ.configs.WEB_URL + "/*" }, function (tabs) {
-    // Send message to each opened tab which matches the url
-    for (let tabIndex = 0; tabIndex < tabs.length; tabIndex++) {
-      chrome.tabs.sendMessage(tabs[tabIndex].id, messageObject, callback);
-    }
+  BG.Methods.getAppTabs().then((tabs) => {
+    tabs.forEach((tab) => {
+      chrome.tabs.sendMessage(tab.id, messageObject, callback);
+    });
   });
 };
 
 BG.Methods.getAppTabs = () => {
   return new Promise((resolve) => {
-    chrome.tabs.query({ url: RQ.configs.WEB_URL + "/*" }, (tabs) => {
-      if (tabs.length === 0) {
-        BG.isAppOnline = false;
-      }
-      resolve(tabs);
+    let appTabs = [];
+
+    RQ.Utils.getAllSupportedWebURLs().forEach((webURL) => {
```

Comment:
`forEach` don't wait for the promise to get resolved. Can we use `for of` loop instead?

Reply:
Ya, my bad. Your [suggestion](https://github.com/requestly/requestly/pull/690/files#diff-5db768afe901c579de3db3cd09570b624ac26eac9f63c4071ae72411df2df116R823) looks good.